backend/product/views.py

```diff
@@ -112,3 +113,183 @@ def add_product(request):
+    if(vendor is None or not vendor.is_verified):
+        return HttpResponse("Vendor authentication failed", status=401)
+    message = ""
+    try:
+        product_id = request.POST["id"]
+    except Exception:
+        return HttpResponse("Product id not specified", status=400)
+    try:
+        product = Product.objects.get(id=product_id)
+    except Exception:
+        return HttpResponse("There is no such product with the id", status=400)
+    if(vendor != product.vendor):
+        return HttpResponse("You cannot edit products of other vendors", status=401)
+    try:
+        category = Category.objects.get(name=request.POST["category"])
+        product.category = category
+        message += "Category, "
+    except Exception:
+        pass
+    try:
+        name = request.POST["name"]
+        product.name = name
+        message += "name, "
+    except KeyError:
+        pass
+    try:
+        brand = request.POST["brand"]
+        product.brand = brand
+        message += "brand, "
+    except KeyError:
+        pass
+    try:
+        description = request.POST["description"]
+        product.description = description
+        message += "description, "
+    except KeyError:
+        pass
+
+    try:
+        stock = int(request.POST["stock"])
+        product.stock = stock
+        message += "stock, "
+    except (KeyError, ValueError):
+        pass
+    try:
+        price = float(request.POST["price"])
+        product.price = price
+        message += "price "
+    except (KeyError, ValueError):
+        pass
+    try:
+        product.save()
+        message += "fields updated."
+    except Exception:
+        return HttpResponse("Product couldn't be created with current params", status=400)
+    try:
+        form = ImageUploadForm(request.POST, request.FILES)
+        if form.is_valid():
+            image = Image.objects.create(product=product)
+            image.photo = form.cleaned_data["photo"]
+            image.save()
+            message += " Photo uploaded."
+    except Exception:
+        pass
+    return HttpResponse(message)
+
+@authentication_classes([SessionAuthentication, BasicAuthentication])
+@permission_classes((IsAuthenticated,))
+@api_view(['DELETE'])
+def delete_product(request):
+    """Delete product with given parameters when DELETE request is made."""
+    vendor = get_vendor_from_request(request)
+    if(vendor is None or not vendor.is_verified):
+        return HttpResponse("Vendor authentication failed", status=401)
+    try:
+        product_id = request.POST["id"]
+    except KeyError:
+        return HttpResponse("Product id not specified", status=400)
+    try:
+        product = Product.objects.get(id=product_id)
+    except Exception:
+        return HttpResponse("There is no such product with the id", status=400)
+    if(vendor != product.vendor):
+        return HttpResponse("You cannot delete products of other vendors", status=401)
+    else:
+        product.delete()
```

Comment:
I think this endpoint doesn't delete the image files. It updates the table though.